mgmt2/rpc/handlers/common/convert.h

```diff
@@ -0,0 +1,121 @@
+/* @file
+   @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+#pragma once
+
+#include <iostream>
+#include <exception>
+
+#include <yaml-cpp/yaml.h>
+
+#include "records/I_RecCore.h"
+#include "records/P_RecCore.h"
+
+#include "shared/overridable_txn_vars.h"
+
+namespace record::field_names
+{
+static constexpr auto Name{"name"};
```

Comment:
I think these should be `std::string` - AFAICT if not, then the value will be converted to `std::string` for comparison, making for a lot of unnecessary allocations.

Reply:
It should call the right overload for `operator==`
```
template< class CharT, class Traits, class Alloc >
bool operator==( const std::basic_string<CharT,Traits,Alloc>& lhs,
                 const CharT* rhs );
```
I'll make sure this is the case.